homework9/hw9task1/iterators_1/merge_sorted_files.py

```diff
@@ -11,16 +11,34 @@
 #>>> list(merge_sorted_files(["file1.txt", "file2.txt"]))
 [1, 2, 3, 4, 5, 6]
 """
-from heapq import merge
 from pathlib import Path
 from typing import Iterator, List, Union
 
 
-def fetching_data(file_path: str):
+def sorting_data(*iters: Iterator):
```

Comment:
Let's think about the different algorithms for merging. 

For example, let's pretend that we have `lists`, what will be the simplest way to merge them.  Describe it in that comment.

Reply:
I suppose the simplest way of merging lists:
`    i, j = 0, 0`
`    result_list = []`
`    while i < len(list1) and j < len(list2):`
`        if list1[i] < list2[j]:`
`            result_list.append(list1[i])`
`            i += 1`
`        else:`
`            result_list.append(list2[j])`
`            j += 1`
`    if i == len(list1)-1 or j == len(list2)-1:`
`        result_list += list1[i:]`
`        result_list += list2[j:]`
`    return result_list`

Reply:
Let's assume that you need to merge multiple lists at once.

Reply:
`def merge_lists(*lists):
    temp_list = []
    result_list = lists[0]
    for l in range(1, len(lists)):
        i, j = 0, 0
        list1 = result_list
        list2 = lists[l]
        while i < len(list1) and j < len(list2):
            if list1[i] < list2[j]:
                temp_list.append(list1[i])
                i += 1
            else:
                temp_list.append(list2[j])
                j += 1
        if i == len(list1) - 1 or j == len(list2) - 1:
            temp_list.extend(list1[i:])
            temp_list.extend(list2[j:])

        result_list = temp_list[:]
        temp_list.clear()
    return result_list`

Reply:
I don`t know how to wrap code here with left spaces in every line.
I write like
'         return smthing`
But it save like 
'return smthing` wihout space before

Reply:
> I don`t know how to write code here with left spaces in every line.

Use code button on multiline (it has different behavior than on single line).

And of cause google for markdown syntax:
https://guides.github.com/features/mastering-markdown/

Reply:
@Cjkjvfnby have you seen this?
https://github.com/IuliiaLanchuk/hometasksEpam/pull/31#discussion_r611233029

Reply:
It is not something I want to get.  Try to get inspiration in the merge-sort algorithm.